diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter5/OverloadingArgumentsFunction.kt

```diff
@@ -36,15 +38,62 @@ class OverloadingArgumentsFunction(configRules: List<RulesConfig>) : DiktatRule(
             .asSequence()
             .filter { it.elementType == FUN }
             .map { it.psi as KtFunction }
-            .filter { it.nameIdentifier!!.text == funPsi.nameIdentifier!!.text && it.valueParameters.containsAll(funPsi.valueParameters) }
+            .filter { it.isOverloadedBy(funPsi) }
+            .filter { it.hasSameModifiers(funPsi) }
             .filter { funPsi.node.findChildBefore(IDENTIFIER, TYPE_REFERENCE)?.text == it.node.findChildBefore(IDENTIFIER, TYPE_REFERENCE)?.text }
             .filter { funPsi.node.findChildAfter(IDENTIFIER, TYPE_REFERENCE)?.text == it.node.findChildAfter(IDENTIFIER, TYPE_REFERENCE)?.text }
             .toList()
+
         if (allOverloadFunction.isNotEmpty()) {
             WRONG_OVERLOADING_FUNCTION_ARGUMENTS.warn(configRules, emitWarn, isFixMode, funPsi.node.findChildByType(IDENTIFIER)!!.text, funPsi.startOffset, funPsi.node)
         }
     }
 
+    /**
+     * We can raise errors only on those methods that have same modifiers (inline/public/etc.)
+     */
+    private fun KtFunction.hasSameModifiers(other: KtFunction) =
+        this.modifierList?.node?.getChildren(KtTokens.MODIFIER_KEYWORDS)
+            ?.map { it.text }
+            ?.sortedBy { it } ==
+                other.modifierList?.node?.getChildren(KtTokens.MODIFIER_KEYWORDS)
```

Comment:
maybe extract to separate method:
```suggestion
private fun KtFunction.getOrderedModifierNames() = modifierList
    ?.node
    ?.getChildren(KtTokens.MODIFIER_KEYWORDS)
    ?.map { it.text }
    ?.sortedBy { it }
```